backend/backend/core/ollama.py

```diff
@@ -1,33 +1,53 @@
 import httpx
 
-from langchain_ollama import ChatOllama
+from pydantic.json_schema import JsonSchemaValue
+from typing import Literal, Union
 
-from utils.logger import logger
+from langchain_ollama import ChatOllama
 
 from config import settings
+from utils.logger import logger
 
 
 class OllamaBackend(object):
     def __init__(self):
         self._pull_models()
 
     def _pull_models(self):
-        logger.info(f"Pulling Ollama model '{settings.OLLAMA_MODEL}' at endpoint '{
-                    settings.OLLAMA_ENDPOINT}' ...")
+        logger.info(
+            f"Pulling Ollama model '{settings.OLLAMA_MODEL}' at endpoint '{
```

Comment:
The `logger.info` function is used for both logging and printing. Consider using separate functions for these purposes.